File: ruby/lib/dogma/longitude.rb
# This is free and unencumbered software released into the public domain.

require_relative 'angle'

##
# A longitude.
#
# @see https://dogma.dev/Longitude
class Dogma::Longitude < Dogma::Angle
  MIN_DEGREES = -180
  MAX_DEGREES = 180

  ##
  # @param  [Float, #to_f] degrees (-180..180)
  # @return [void]
  # @raise  [ArgumentError] if `degrees` is out of bounds
  def initialize(degrees)
    degrees = degrees.to_f
    raise ArgumentError, "Longitude bounds are ±180°, but got #{degrees}°" if degrees.abs > MAX_DEGREES
    super(degrees: degrees)
  end
end # Dogma::Longitude

File: python/src/dogma/longitude.py
# This is free and unencumbered software released into the public domain.

"""The dogma.Longitude class."""

from .angle import Angle

# See: https://dogma.dev/Longitude
class Longitude(Angle):
    """A longitude."""

    MIN_DEGREES = -180
    MAX_DEGREES = 180

    def __init__(self, degrees: float) -> None:
        degrees = float(degrees)
        if abs(degrees) > Longitude.MAX_DEGREES:
            raise ValueError(f"Longitude bounds are ±180°, but got {degrees}°")
        super().__init__(degrees=degrees)
